### PR TITLE
Added `WUPSConfigAPI_GetMenuOpen()` to allow plugins to check if the menu is open.

### PR DESCRIPTION
**Pull request by dkosmari**
This new function allows a plugin to check if the config menu is open. It's now possible for a plugin to change behavior when the user is trying to interact with the menu.

### CODE REVIEW

File: source/config/WUPSConfigAPI.cpp [hunk 1]
```diff
@@ -311,7 +311,16 @@ namespace WUPSConfigAPIBackend {
         return WUPSCONFIG_API_RESULT_SUCCESS;
     }
 
+    WUPSConfigAPIStatus WUPSConfigAPI_GetMenuOpen(BOOL *out) {
```
Comment (Maschell) [inline source/config/WUPSConfigAPI.cpp:314]: How about something like this?

`WUPSConfigAPIStatus WUPSConfigAPI_Menu_GetStatus(WUPSConfigAPIMenuStatus *out)`

with defined in WUPS?

```
typedef enum WUPSConfigMenuStatus {
    WUPSCONFIG_API_MENU_STATUS_CLOSED  = 0,
    WUPSCONFIG_API_MENU_STATUS_OPENED = 1,
} WUPSConfigMenuStatus ;
```

Comment (dkosmari) [reply source/config/WUPSConfigAPI.cpp]: Updated both PRs with the new enum and new function name.

**Comment by Maschell** [comment]
Please fix the formatting and update the Dockerfile to use `COPY --from=ghcr.io/wiiu-env/wiiupluginsystem:0.8.2-dev-20241128-1ac579a`

**Comment by dkosmari** [comment]
Replaced by #89.

**Pull request closed without merging**